front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorForm.tsx

```diff
@@ -36,6 +40,28 @@ type RollingStockParametersProps = {
   isAdding?: boolean;
 };
 
+export function modifyRollingStockElectricalValues(
+  currentRollingStockValues: RollingStockParametersValues,
+  prevRsEffortCurve: RollingStock['effort_curves'] | null,
+  currentRsEffortCurve: RollingStock['effort_curves'] | null
+) {
+  if (!isElectric(prevRsEffortCurve) && isElectric(currentRsEffortCurve)) {
+    return {
+      ...currentRollingStockValues,
+      electricalPowerStartupTime: DEFAULT_ELEC_POWER_STARTUP_TIME,
+      raisePantographTime: DEFAULT_RAISE_PANTOGRAPH_TIME,
+    };
+  }
+  if (isElectric(prevRsEffortCurve) && !isElectric(currentRsEffortCurve)) {
+    return {
+      ...currentRollingStockValues,
+      electricalPowerStartupTime: null,
+      raisePantographTime: null,
+    };
+  }
+  return currentRollingStockValues;
+}
```

Comment:
```suggestion
export function modifyRollingStockElectricalValues(
  currentRollingStockValues: RollingStockParametersValues,
  isPrevRsEffortCurve: boolean,
  isCurrentRsEffortCurve: boolean
) {
  if (!isPrevRsEffortCurve && isCurrentRsEffortCurve) {
    return {
      ...currentRollingStockValues,
      electricalPowerStartupTime: DEFAULT_ELEC_POWER_STARTUP_TIME,
      raisePantographTime: DEFAULT_RAISE_PANTOGRAPH_TIME,
    };
  }
  if (isPrevRsEffortCurve && !isCurrentRsEffortCurve) {
    return {
      ...currentRollingStockValues,
      electricalPowerStartupTime: null,
      raisePantographTime: null,
    };
  }
  return currentRollingStockValues;
}
```
We can maybe check if the rs has an electric mode before calling this function to avoid calling `isElectric()` multiple times ?

Reply:
Ok I can create 2 variables, but I don’t really want to change the function signature, I think it’s better hidden inside the function.